File: software/examples/python/example_callback.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "abc" # Change to your UID

NUM_LEDS = 16

r = [0]*NUM_LEDS
g = [0]*NUM_LEDS
b = [0]*NUM_LEDS
r_index = 0

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_led_strip import LEDStrip

# Frame rendered callback, is called when a new frame was rendered
# We increase the index of one blue LED with every frame
def cb_frame_rendered(led_strip, length):
    global r_index
    b[r_index] = 0
    if r_index == NUM_LEDS-1:
        r_index = 0
    else:
        r_index += 1

    b[r_index] = 255

    # Set new data for next render cycle
    led_strip.set_rgb_values(0, NUM_LEDS, r, g, b)

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    led_strip = LEDStrip(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Set frame duration to 50ms (20 frames per second)
    led_strip.set_frame_duration(50)

    # Register frame rendered callback to function cb_frame_rendered
    led_strip.register_callback(led_strip.CALLBACK_FRAME_RENDERED, 
                                lambda x: cb_frame_rendered(led_strip, x))

    # Set initial rgb values to get started
    led_strip.set_rgb_values(0, NUM_LEDS, r, g, b)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'oud'; // Change to your UID

var NUM_LEDS = 16;
var r = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
var g = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
var b = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
var rIndex = 0;

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var ls = new Tinkerforge.BrickletLEDStrip(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Set frame duration to 50ms (20 frames per second)
        ls.setFrameDuration(50);
        // Set initial rgb values to get started
        ls.setRGBValues(0, NUM_LEDS, r, g, b);
    }
);

// Register frame rendered callback
ls.on(Tinkerforge.BrickletLEDStrip.CALLBACK_FRAME_RENDERED,
    // Frame rendered callback, is called when a new frame was rendered
    // We increase the index of one blue LED with every frame
    function(len) {
        b[rIndex] = 0;
        if(rIndex === NUM_LEDS-1) {
            rIndex = 0;
        }
        else {
            rIndex += 1;
        }
        b[rIndex] = 255;
        // Set new data for next render cycle
        ls.setRGBValues(0, NUM_LEDS, r, g, b)
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
